process/block/metablock.go

```diff
@@ -523,9 +525,20 @@ func (mp *metaProcessor) indexBlock(
 		return
 	}
 
-	go mp.core.Indexer().SaveBlock(body, metaBlock, txPool, signersIndexes)
+	go mp.core.Indexer().SaveBlock(body, metaBlock, txPool, signersIndexes, notarizedHeadersHashes)
 
-	saveRoundInfoInElastic(mp.core.Indexer(), mp.nodesCoordinator, core.MetachainShardId, metaBlock, lastMetaBlock, signersIndexes)
+	indexRoundInfo(mp.core.Indexer(), mp.nodesCoordinator, core.MetachainShardId, metaBlock, lastMetaBlock, signersIndexes)
+
+	if metaBlock.GetNonce() == 1 {
```

Comment:
is there a way to merge the 2 ifs: this one and the one below?

Reply:
```
if metaBlock.GetNonce() != 1 && !metaBlock.IsStartOfEpochBlock(){
		return
}
```

done.

Reply:
ok